config/pmd.xml

```diff
@@ -104,66 +129,82 @@
-           validateCli is not reasonable to split as encapsulation of logic will be damaged
-           getDetails - huge Switch, it has to be monolithic
-           JavadocMethodCheck, JavadocStyleCheck, JavadocUtils.getJavadocTags() - deprecated
-           FinalLocalVariableCheck.visitToken() - it is just big switch , not could be done
-           GenericWhitespaceCheck.processSingleGeneric() - it contains a lot of similar to each other and very simple condition
-           VariableDeclarationUsageDistanceCheck, CustomImportOrderCheck - it is not OK to have such a complicated logic need to be refactored.
-      -->
-      <property name="violationSuppressXPath" value="//DeclarationOrder[@Name='visitToken'] and //MethodDeclaration[@Name='validateCli' and ../../..[@Image='Main']] | //MethodDeclaration[@Name='processSingleGeneric' and ../../..[@Image='GenericWhitespaceCheck']] | //MethodDeclaration[@Name='visitToken' and ../../..[@Image='FinalLocalVariableCheck']] | //MethodDeclaration[@Name='getJavadocTags' and ../../..[@Image='JavadocUtils']] | //MethodDeclaration[@Name='getDetails' and ../../..[@Image='RightCurlyCheck']] | //ClassOrInterfaceDeclaration[@Image='JavadocMethodCheck' or @Image='JavadocStyleCheck' or @Image='VariableDeclarationUsageDistanceCheck' or @Image='CustomImportOrderCheck']"/>
-    </properties>
-  </rule>
-  <rule ref="rulesets/java/codesize.xml/NPathComplexity">
-    <properties>
-      <!-- JavadocMethodCheck - deprecated
-           WhitespaceAroundCheck.isNotRelevantSituation - npath = 216, but additional extraction
-                                of expression to separate method does not make sense as it will
-                                damage encapsulation of that check.
-      -->
-      <property name="violationSuppressXPath" value="//MethodDeclaration[@Name='isNotRelevantSituation' and ../../..[@Image='WhitespaceAroundCheck']] | //MethodDeclaration[@Name='validateCli' and ../../..[@Image='Main']] | //ClassOrInterfaceDeclaration[@Image='JavadocMethodCheck' or @Image='JavadocStyleCheck']"/>
+      <!-- RequireThisCheck, VariableDeclarationUsageDistanceCheck, XdocsPagesTest,
+             JavadocMethodCheck and CommentsIndentationCheck are big classes and should not
+             be split without a good reason.
+           Main.validateCli is not reasonable to split as encapsulation of the logic will be
+             damaged.
+           *.visitToken are a big SWITCH block with a number of IF blocks.
+             If we split the block to several methods it will damage the readability.
+           JavadocStyleCheck is deprecated.
+           SuppressWarningsHolder.getAcceptableParent, FallThroughCheck.isTerminated,
+             ElementNode.iterateAxis, RootNode.iterateAxis are also huge switches,
+             they has to be monolithic.
+           CustomImportOrderCheck.finishImportList, CustomImportOrderCheck.getImportGroup ???
+           Checker.processFiles ???
+           CheckUtils.parseDouble ??? -->
```

Comment:
???s have not been resolved

Reply:
Done. `CheckUtils.parseDouble` is refactored as #5709 
`CustomImportOrderCheck.finishImportList`, `CustomImportOrderCheck.getImportGroup` and
`Checker.processFiles` are good enough. No reason to refactor these methods.